novice-py-getting-started.Rmd

```diff
@@ -0,0 +1,335 @@
+
+## Running Code via Scripts
+
+Can you imagine a sitution where continuing to type code into the console 
+could become tedious or challenging? 
+
+Here are some examples: 
+
+* Wanting to run the same piece of code many times (you 
+*can* use the up-arrow to see previous commands - try it! - but if you 
+run many commands, your history will quickly become more cluttered. 
+* Needing to re-run a piece of code
+* Most important - when you close Spyder, everything you've typed into the 
+console is gone. If you want to start again, you'll need to re-enter everything. 
+
+So we're motivated to find a way to write and run code where we're keeping 
+a record as we go along, and it is easy to run code again, in chunks. 
+
+The answer to these challenges is to move to the left pane of the Spyder 
+IDE. This is a script - a text file with Python code. Try copying in some 
+of the commands we've already run: 
+
+```{python}
+TBD
+```
+
+Now these commands are written in the script, but nothing has happened yet. 
+In order to run the commands from the script (just like we did in the console), 
+we'll have to use various run commands. You can see these under the "Run" menu; 
+but typically you'll want to learn at least one or two keyboard shortcuts so 
+it becomes quick and easy to run code that's written in a script. Try "running" 
+the lines in your script. You should see their output in the console on the 
+bottom right. 
+
+Sometimes, when you want to experiment or check something, it can make sense 
+to write and run Python code in the console. However, most of the time, you'll 
+want to write and run your code from a script.  Using 
+scripts has the benefit of saving your work, while also being able to run 
+the code just like in the console. The script then has the added benefit of 
+being easy to share and runnable outside of the IDE. 
+
+We've now told the computer what to do by using Python code. We've run that 
+code in two different ways within our "workbench" - the Spyder IDE. Let's 
+see what else we can use in this environment to help us. 
+
+## Variables
+
+Besides running functions that *do* something (as above), we'll also want 
+to use Python to keep track of information that we're using throughout 
+our analysis. We save that information by giving it a name (called a *variable*), 
+using the name, an equals sign (`=`) and then the information to save. 
+For example: 
+
+```{python}
+message = "It is a far, far better thing that I do, than I have ever done"
+name_length = len("Sydney Carton")
+```
+
+These *variables* can then be used later in functions like `print()`: 
+
+```{python}
+print("My message:", message)
+print("Letters in name:", name_length)
+```
+
+On the top right pane, there's a 
+tab that says "Variable Explorer". If you click on that, you'll see the 
+variables you just created, with some additional information about the *Type* 
+of information - `str` (string), `int` (integer), `float` (decimal point 
```

Comment:
errr. float? why's it called a float? what's it floating on? or is it some sort of life preserver to help those who feel like they're drowning?